src/routes/safe/components/Transactions/TxList/TxDetails.tsx

```diff
@@ -127,38 +128,62 @@ export const TxDetails = ({ transaction }: TxDetailsProps): ReactElement => {
     )
   }
 
+  const noData = !data.txData || (!data.txData.dataDecoded && !data.txData.hexData)
   return (
     <TxDetailsContainer ownerRows={hasModule ? 3 : 2}>
-      <div>
-        <div
-          className={cn('tx-details', {
-            'no-padding': isMultiSendTxInfo(data.txInfo),
-            'not-executed': !data.executedAt,
-            'will-be-replaced': willBeReplaced,
-          })}
-        >
-          <TxDataGroup txDetails={data} />
-        </div>
-        {isMultiSend && getModuleDetails()}
-        <div className={cn('tx-summary', { 'will-be-replaced': willBeReplaced })}>
-          <TxSummary txDetails={data} />
-        </div>
-      </div>
-      <div>
-        {!isMultiSend && getModuleDetails()}
-        <div
-          className={cn('tx-owners', {
-            'will-be-replaced': willBeReplaced,
-          })}
-        >
-          <TxOwners txDetails={data} isPending={isPending} />
-        </div>
-        {!isPending && !data.executedAt && txLocation !== 'history' && !!currentUser && (
-          <div className={cn('tx-details-actions', { 'will-be-replaced': willBeReplaced })}>
-            <TxExpandedActions transaction={transaction} />
-          </div>
+      <div className={cn('tx-data', { 'no-owners': noOwners, 'no-data': noData })}>
+        {isMultiSend && (
```

Comment:
These are both quite chunky bits of code that are based on `isMultiSend`. It might be clearer to abstract this. It could still remain in this file.

Reply:
moved it to a function but it becomes a bit harder to read IMO